### DISCUSSION TITLE
Container Sandboxes still have problems after changing dependencies :(

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [codesandbox/codesandbox-client#5395] [Q&A] [answered]
Here is an example: https://codesandbox.io/s/modifier-error-stacktrace-test-zb5bl?file=/package.json

I've refreshed the page, and tried clicking restart sandbox/server, but I'm stuck in this state: 
![image](https://user-images.githubusercontent.com/199018/105616292-ecdeaa00-5da3-11eb-84cb-8d36c976e912.png)

I think the _only_ way out of this is if I leave the sandbox alone for 30 minutes (and if no one visits the URL for 30 minutes), so that the container is totally torn down. :-\ 

Questions:
 - Why does this happen?
 - Are there any secret commands I can run from the console to reset things, since all the buttons are disabled?

**Comment by NullVoxPopuli** [comment]
It seems something else has gone terribly wrong. Nothing will recover the sandbox

**Reply by lbogdan** [reply]
Hey @NullVoxPopuli ,

There was an issue with the persistent storage for that sandbox, it should be fixed now. However, I get a runtime error when starting the sandbox:
```
Missing yarn packages:
Package:  ember-could-get-used-to-this
  * Specified: *
  * Installed: (not installed)

Run `yarn` to install missing dependencies.
```

**Reply by NullVoxPopuli** [reply]
Thanks

**Comment by NullVoxPopuli** [comment]
I thought the sandbox detected package changes and ran yarn for me. 🤔 

I'll run yarn manually and see what happens

**Comment by NullVoxPopuli** [comment]
Something is still very wrong. Yarn. At runtime, thinks I don't have the new package, yet, when I just cut 'yarn', it says 'up to date', and when I try to install the new package via CLI, yarn... Can't find the list of packages?

![Screenshot_20210125-080843](https://user-images.githubusercontent.com/199018/105710049-91680580-5ee4-11eb-9b6f-cdbbcd83fafd.png)

Also, ignore the triple dash, it seems there is a gross rendering disconnect between what in typed and what is displayed. When backspacing, I need to delete everything twice for some reason... But that's a separate issue

**Reply by lbogdan** [reply]
Never had any reported issues regarding `yarn`, so I'd say there's something else at work here.

**Reply by NullVoxPopuli** [reply]
Yeah, the error is coming from a dependency chuckle library that analyzes node modules for correctness. So, something in misreporting to that library, afaict. Never had this problem locally

**Comment by NullVoxPopuli** [comment]
Does codesandbox override any node utils? These in particular? https://github.com/quaertym/ember-cli-dependency-checker/blob/master/lib/dependency-checker.js#L7

**Reply by lbogdan** [reply]
We don't, we just run the main script the same as you'd run it locally.

**Reply by NullVoxPopuli** [reply]
Iiinnnteresting. I'll keep digging

**Comment by NullVoxPopuli** [comment, marked as answer]
@lbogdan and I chatted on discord, and they found that I had an extra space before the package name in package.json.

Omg

So, " package" in dependencies was causing all this.
Oof.

So much oof
